java-formats/kml-format_kml/src/com/dataiku/dss/utils/KMLParser.java

```diff
@@ -0,0 +1,145 @@
+package com.dataiku.dss.utils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.dataiku.dip.datalayer.ColumnFactory;
+import com.dataiku.dip.datalayer.ProcessorOutput;
+import com.dataiku.dip.datalayer.Row;
+import com.dataiku.dip.datalayer.RowFactory;
+import com.dataiku.dip.utils.DKULogger;
+import org.apache.commons.lang.StringUtils;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.dataiku.dip.shaker.types.GeoPoint.Coords;
+
+public class KMLParser {
+
+    public KMLParser() {}
+
+    public Element getFirstNodeByTagName(Element parent, String name){
+        for (int i=0; i < parent.getChildNodes().getLength(); i++){
+            logger.info("Child "+i+":"+parent.getChildNodes().item(i));
+        }
+        NodeList nl = parent.getElementsByTagName(name);
+        if (nl.getLength()==0) return null;
+        else return (Element)nl.item(0);
+    }
+
+    private void putAttrValueIfExists(ColumnFactory cf, Row r, String columnName, Element e, String attrName) {
+        String attrValue = e.getAttribute(attrName);
+        if (!StringUtils.isBlank(attrValue)) {
+            r.put(cf.column(columnName), attrValue);
+        }
+    }
+
+    private void putContentIfExistsInChild(ColumnFactory cf, Row r, String columnName, Element e, String childNodeName) {
+        Node childNode = getFirstNodeByTagName(e, childNodeName);
+        if (childNode != null) {
+            String txt = childNode.getTextContent();
+            if (txt != null) {
+                r.put(cf.column(columnName), txt);
+            }
+        }
+    }
+
+    private void parsePlacemark(Node node, ProcessorOutput out, ColumnFactory cf, RowFactory rf) throws Exception {
+        Element e = (Element)node;
```

Comment:
Test if node can be casted to Element
```suggestion
        if( node instanceof Element) {
              Element e = (Element)node;
              ...
       }
```